WordPress/Jetpack/Classes/ViewRelated/WordPress-to-Jetpack Migration/Common/MigrationEmailService.swift

```diff
@@ -0,0 +1,97 @@
+import Foundation
+
+final class MigrationEmailService {
+
+    // MARK: - Dependencies
+
+    private let api: WordPressComRestApi
+
+    // MARK: - Init
+
+    init(api: WordPressComRestApi) {
+        self.api = api
+    }
+
+    convenience init(account: WPAccount) {
+        self.init(api: account.wordPressComRestV2Api)
+    }
+
+    /// Convenience initializer that tries to set the `WordPressComRestApi` property, or fails.
+    convenience init() throws {
+        do {
+            let context = ContextManager.shared.mainContext
+            guard let account = try WPAccount.lookupDefaultWordPressComAccount(in: context) else {
+                throw MigrationError.accountNotFound
+            }
+            self.init(api: account.wordPressComRestV2Api)
+        } catch let error {
+            DDLogError("[\(MigrationError.domain)] Object instantiation failed: \(error)")
+            throw error
+        }
+    }
+
+    // MARK: - Methods
+
+    func sendMigrationEmail() async throws {
+        do {
+            WPAnalytics.track(.migrationEmailTriggered)
+            let response = try await sendMigrationEmail(path: Endpoint.migrationEmail)
+            if !response.success {
+                throw MigrationError.unsuccessfulResponse
+            }
+            WPAnalytics.track(.migrationEmailSent)
+        } catch let error {
+            let properties = trackEventProperties(from: error)
```

Comment:
I like this! It worked well by providing a descriptive error when I tested without a network connection.